tide/core/PowerTimer.cpp

```diff
@@ -0,0 +1,100 @@
+/*                                                                   */
+/*   2. Redistributions in binary form must reproduce the above      */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer in the documentation and/or other materials       */
+/*      provided with the distribution.                              */
+/*                                                                   */
+/*    THIS  SOFTWARE IS PROVIDED  BY THE  UNIVERSITY OF  TEXAS AT    */
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of Ecole polytechnique federale de Lausanne.          */
+/*********************************************************************/
+
+#include "PowerTimer.h"
+
+#include "log.h"
+
+PowerTimer::PowerTimer()
+{
+}
+
+PowerTimer::PowerTimer(const int timeout)
+    : _timeout(timeout * 60000) // from ms to minute
+    , _countDownTimer(new QTimer())
+    , _inactivityTimer(new QTimer())
+
+{
+    connect(_inactivityTimer.get(), &QTimer::timeout, [this]() {
+        if (!_countDownTimer->isActive())
+        {
+            _active = true;
+            _countDownTimer->start(_countdownTimeout);
+            emit updated(shared_from_this());
+        }
+    });
+    connect(_countDownTimer.get(), &QTimer::timeout,
+            [this]() { emit poweroff(); });
+}
+
+int PowerTimer::getCountdownTimeout()
+{
+    return _countdownTimeout;
+}
+
+bool PowerTimer::isActive()
+{
+    return _active;
+}
+
+void PowerTimer::start()
+{
+    _inactivityTimer->start(_timeout);
+}
+
+void PowerTimer::stop()
+{
+    _active = false;
+    emit updated(shared_from_this());
+    _inactivityTimer->stop();
+    _countDownTimer->stop();
+}
+
+void PowerTimer::reset()
+{
+    _inactivityTimer->start();
```

Comment:
not using _timeout here? why not do:
_inactivityTimer->setInterval(int msec);
_inactivityTimer->setSingleShot(true);
in the class constructor?